### PR TITLE
General fixes v2 1 0

### PR DESCRIPTION
**Pull request by nottoast**

### CODE REVIEW

File: backup_database.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+curl https://spoonstracker.firebaseio.com/.json?format=export /home/ubuntu/backups/spoonstracker_db_backup_$(date -d "today" +"%Y%m%d%H%M").json
```
Comment (matkastner) [inline backup_database.sh]: I'd like this to be aws lambdas or something at some point but this is fine for now

File: src/spoons-tracker-admin/spoons-tracker-admin.html [hunk 2]
```diff
@@ -49,10 +49,12 @@
     </style>
 
     <firebase-auth id="auth" user="{{user}}" provider="facebook"></firebase-auth>
-    <firebase-document path="/users/[[user.uid]]/isAdmin" data={{isAdmin}}></firebase-document>
+    <firebase-document path="/users/[[user.uid]]" data={{userDetails}}></firebase-document>
+    <firebase-document id="properties" path="/properties/admin/deleteButton" data="{{deleteProperty}}"></firebase-document>
 
     <firebase-query id="query" path="/userPubs" data="{{pubs}}"></firebase-query>
 
+    <!--<template is="dom-if" if="{{userDetails.isAdmin}}">-->
```
Comment (matkastner) [inline src/spoons-tracker-admin/spoons-tracker-admin.html]: Why is this commented out?

File: src/spoons-tracker-map/spoons-tracker-map.html [hunk 3]
```diff
@@ -61,18 +70,72 @@
...
+
+      _checkLength: function(length) {
+        if(length > 0) {
+          return true;
+        } else {
+          return false;
+        }
+      },
+
+      _test: function() {
```
Comment (matkastner) [inline src/spoons-tracker-map/spoons-tracker-map.html]: Remove this test function

File: src/spoons-tracker-map/spoons-tracker-map.html [hunk 4]
```diff
@@ -61,18 +70,72 @@
...
+        console.log('test');
       }
+
     });
+
+    function toggleVisit(key) {
+      firebase.database().ref('users/' + firebase.auth().currentUser.uid + '/visited/' + key).once("value",function(snapshot){
```
Comment (matkastner) [inline src/spoons-tracker-map/spoons-tracker-map.html]: Pretty sure you can just do `this.set('this.visits.' + key, null)` and set visits to `notify: true` and `readOnly: false` beacuse this array is linked to firebase already.. no need to do it manually.

Comment (matkastner) [inline src/spoons-tracker-map/spoons-tracker-map.html]: i'll test this myself when i get some time but its the same as pub-list-item

File: backup_database.sh (hunk 1 above)
Comment (nottoast) [inline backup_database.sh]: Yeah that sounds good, just a temp solution for the time being

File: src/spoons-tracker-admin/spoons-tracker-admin.html (hunk 2 above)
Comment (nottoast) [inline src/spoons-tracker-admin/spoons-tracker-admin.html]: I actually fixed this in a later change, now non-admin users get redirected off the admin page and the firebase rules mean they can't make any changes to pubs anyway

File: src/spoons-tracker-map/spoons-tracker-map.html (hunk 3 above)
Comment (nottoast) [inline src/spoons-tracker-map/spoons-tracker-map.html]: It goes in a later change

File: src/spoons-tracker-map/spoons-tracker-map.html (hunk 4 above)
Comment (nottoast) [inline src/spoons-tracker-map/spoons-tracker-map.html]: Yeah think you can.. my issue was I couldn't get a button in a map marker to call a polymer function, so had to make it a bog standard JS one. I searched for quite a while but couldn't get anywhere, think there is some issue having data binding in a map marker for some reason. If you can figure it out then what you said would be a better solution

Comment (matkastner) [inline src/spoons-tracker-map/spoons-tracker-map.html]: thats fine for now

**Pull request merged**